front/components/items/Appointment.tsx

```diff
@@ -6,35 +6,55 @@ import { DayPicker } from "react-day-picker";
 import { Grid } from "react-loader-spinner";
 
 const Appointment = () => {
-    const therapist = "Bożena"
-    let [chosenDate, setChosenDate] = useState<Date>();
-    let filters: filters = {};
+    const TEMPtherapist = "Bożena";
+    let rightTime: Date;
+    let queryChosenDate: string | undefined;
+    const filters: filters = {};
     filters.therapist = {
         first_name: {
-            $eq: therapist
+            $eq: TEMPtherapist
         }
-    }
-    filters.date = {
-        $eq: chosenDate
-    }
-    const [isLoading, setIsLoading] = useState<boolean>(true);
-    const [appointmentsTherapist, setAppointentsTherapist] = useState<Array<any>>();
-    const [appointmentsDate, setAppointentsDate] = useState<Array<any>>();
+    };
+    let [chosenDate, setChosenDate] = useState<Date>();
+    const [isLoading, setIsLoading] = useState(true);
+    const [appointmentsTherapist, setAppointmentsTherapist] = useState<appointmentQueryType[]>();
+    const [appointmentsDate, setAppointmentsDate] = useState<appointmentQueryType[]>();
     useEffect(() => {
-        strapiAppointmentGet(filters.therapist).then((res) => {
-                setAppointentsTherapist(res.data.data);
-                setIsLoading(false);
-            }
-        );
-        strapiAppointmentGet(filters.therapist.date).then((res) => {
-                setAppointentsDate(res.data.data);
+        if (chosenDate) {
+            // changing timezone by reducing by 2h (because of polish timezone (GMT+2))
+            rightTime = new Date(chosenDate?.getTime() - (-120*60*1000));    
+            // changing format of date to YYYY-MM-DD
+            queryChosenDate = rightTime?.toISOString().slice(0, 10);
+            filters.date = {
+                $eq: queryChosenDate
+            };
+        }
+        strapiAppointmentGet(filters).then((res) => {
+                setAppointmentsTherapist(res.data.data);
+                setAppointmentsDate(res.data.data);
                 setIsLoading(false);
             }
         );
     }, [chosenDate]);
-    // TODO: aktualizowanie filtrów z nowymi zmiennymi
-    console.log(chosenDate);
-    console.log("-----------------------");
+    const allDates = appointmentsTherapist?.map(item => item.attributes.date);
+    // deleting repeated dates 
+    /* indexOf is searching the first index numer of current item value and 
+    then compering to the left elements. If they're same, value's true, 
+    if they're reapeted value's false and won't stay in new array
+    e.g "water" is 0, but also 3, 4, 5, so only "0 water" would stay, because 0 doesn't equal 3 */
+    const stringDate = allDates?.filter((item, index) => {return allDates?.indexOf(item) === index});
+    const availableDate = stringDate?.map(item => new Date(item));
+    let allHours = appointmentsDate?.map(item => item.attributes.time);
+    let sortHours = allHours?.sort(function (a, b) {
```

Comment:
I'd name it rather 'sorted' bcs verbals in names sounds more like function names